Makefile

```diff
@@ -1,3 +1,9 @@
+# ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
+ENVTEST_K8S_VERSION = 1.25
```

Comment:
Do we want to be on 1.26 now?

Reply:
With this change, once of the vendor files doesn't compile and some of them need Go version `1.19` as well. What do you suggest now?

Reply:
We can bump to go 1.19 :) If that doesn't work, please post the errors

Reply:
That didn't work. Having this error on 1.18.
[error.txt](https://github.com/openshift/cluster-machine-approver/files/10345521/error.txt)

Reply:
KUBEBUILDER_ASSETS="/home/dodvarka/go/src/cluster-machine-approver/bin/k8s/1.26.0-linux-amd64" ./hack/ci-test.sh
go run ./hack/../vendor/github.com/onsi/ginkgo/v2/ginkgo -v --randomize-all --randomize-suites --keep-going --race --trace --timeout=10m <omitted>
=== RUN   Test_authorizeCSR
=== RUN   Test_authorizeCSR/should_list_cluster-api_machines_in_one_namespace
=== RUN   Test_authorizeCSR/should_list_openshift_machines_in_all_namespaces_when_namespace_is_empty
--- PASS: Test_authorizeCSR (0.01s)
    --- PASS: Test_authorizeCSR/should_list_cluster-api_machines_in_one_namespace (0.00s)
    --- PASS: Test_authorizeCSR/should_list_openshift_machines_in_all_namespaces_when_namespace_is_empty (0.00s)
PASS
Failed to compile controller:

# sigs.k8s.io/controller-runtime/pkg/cache
../../vendor/sigs.k8s.io/controller-runtime/pkg/cache/informer_cache.go:144:9: cannot use i.Informer (variable of type "k8s.io/client-go/tools/cache".SharedIndexInformer) as type Informer in return statement:
	"k8s.io/client-go/tools/cache".SharedIndexInformer does not implement Informer (wrong type for AddEventHandler method)
		have AddEventHandler(handler "k8s.io/client-go/tools/cache".ResourceEventHandler) ("k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration, error)
		want AddEventHandler(handler "k8s.io/client-go/tools/cache".ResourceEventHandler)
../../vendor/sigs.k8s.io/controller-runtime/pkg/cache/informer_cache.go:158:9: cannot use i.Informer (variable of type "k8s.io/client-go/tools/cache".SharedIndexInformer) as type Informer in return statement:
	"k8s.io/client-go/tools/cache".SharedIndexInformer does not implement Informer (wrong type for AddEventHandler method)
		have AddEventHandler(handler "k8s.io/client-go/tools/cache".ResourceEventHandler) ("k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration, error)
		want AddEventHandler(handler "k8s.io/client-go/tools/cache".ResourceEventHandler)


Ginkgo ran 2 suites in 1.791606638s

There were failures detected in the following suites:
  controller ./pkg/controller [Compilation failure]

Test Suite Failed